src/govuk/components/exit-this-page/exit-this-page.mjs

```diff
@@ -0,0 +1,393 @@
+  this.timeoutMessageId = null
+}
+
+/**
+ * Create the <span> we use for screen reader announcements.
+ *
+ * @deprecated Will be made private in v5.0
+ */
+ExitThisPage.prototype.initUpdateSpan = function () {
+  this.$updateSpan = document.createElement('span')
+  this.$updateSpan.setAttribute('role', 'status')
+  this.$updateSpan.className = 'govuk-visually-hidden'
+
+  this.$module.appendChild(this.$updateSpan)
+}
+
+/**
+ * Create button click handlers.
+ *
+ * @deprecated Will be made private in v5.0
+ */
+ExitThisPage.prototype.initButtonClickHandler = function () {
+  // Main EtP button
+  this.$button.addEventListener('click', this.handleClick.bind(this))
+
+  // EtP skiplink
+  if (this.$skiplinkButton) {
+    this.$skiplinkButton.addEventListener('click', this.handleClick.bind(this))
+  }
+}
+
+/**
+ * Create the HTML for the 'three lights' indicator on the button.
+ *
+ * @deprecated Will be made private in v5.0
+ */
+ExitThisPage.prototype.buildIndicator = function () {
+  // Build container
+  // Putting `aria-hidden` on it as it won't contain any readable information
+  this.$indicatorContainer = document.createElement('div')
+  this.$indicatorContainer.className = 'govuk-exit-this-page__indicator'
+  this.$indicatorContainer.setAttribute('aria-hidden', 'true')
+
+  // Create three 'lights' and place them within the container
+  for (var i = 0; i < 3; i++) {
+    var $indicator = document.createElement('div')
+    $indicator.className = 'govuk-exit-this-page__indicator-light'
+    this.$indicatorContainer.appendChild($indicator)
+  }
+
+  // Append it all to the module
+  this.$button.appendChild(this.$indicatorContainer)
+}
+
+/**
+ * Update whether the lights are visible and which ones are lit up depending on
+ * the value of `keypressCounter`.
+ *
+ * @deprecated Will be made private in v5.0
+ */
+ExitThisPage.prototype.updateIndicator = function () {
+  // Show or hide the indicator container depending on keypressCounter value
+  if (this.keypressCounter > 0) {
+    this.$indicatorContainer.classList.add('govuk-exit-this-page__indicator--visible')
+  } else {
+    this.$indicatorContainer.classList.remove('govuk-exit-this-page__indicator--visible')
+  }
+
+  // Turn on only the indicators we want on
+  var $indicators = this.$indicatorContainer.querySelectorAll(
+    '.govuk-exit-this-page__indicator-light'
+  )
+  nodeListForEach($indicators, function ($indicator, index) {
+    $indicator.classList.toggle(
+      'govuk-exit-this-page__indicator-light--on',
+      index < this.keypressCounter
+    )
+  }.bind(this))
+}
+
+/**
+ * Initiates the redirection away from the current page.
+ * Includes the loading overlay functionality, which covers the current page with a
+ * white overlay so that the contents are not visible during the loading
+ * process. This is particularly important on slow network connections.
+ *
+ * @deprecated Will be made private in v5.0
+ */
+ExitThisPage.prototype.exitPage = function (redirectUrl) {
```

Comment:
Just double checking, was `redirectUrl` supposed to be an optional override?

It's not used at the moment

```suggestion
ExitThisPage.prototype.exitPage = function () {
```

Reply:
Ah whoops. This was a carryover from when I was trying out splitting how the URL we're passing to the navigation was defined. I'll get rid of it.